src/index.ts

```diff
@@ -137,9 +137,21 @@ function addCommands(
   tracker: WidgetTracker<IDocumentWidget<EditableCSVViewer>>
 ): void {
   const { commands } = app;
-  const SELECTOR = '.jp-CSVViewer-grid';
+  const GLOBAL_SELECTOR = '.jp-CSVViewer-grid';
+  const BODY_SELECTOR = '.jp-background';
+  const COLUMN_HEADER_SELECTOR = '.jp-column-header';
+  const ROW_HEADER_SELECTOR = '.jp-row-header';
 
-  commands.addCommand(CommandIDs.addRow, {
+  commands.addCommand(CommandIDs.insertRowAbove, {
+    label: 'Add Row',
+    execute: () => {
+      // emit a signal to the EditableDSVModel
+      tracker.currentWidget &&
+        tracker.currentWidget.content.changeModelSignal.emit('add-row');
+    }
+  });
+
+  commands.addCommand(CommandIDs.insertRowBelow, {
     label: 'Add Row',
```

Comment:
Insert Row Below here and for the signal emitted

Reply:
Changes made!